[libpq:x64-windows] build failure

**Host Environment**
- OS: Windows 10
- Compiler: Vcpkg package management program version 2021-11-02-af04ebf6274fd6f7a941bff4662b3955c64f6f42

**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg install libpq:x64-windows

**Failure logs** 
The following packages will be built and installed:
    libpq[core,openssl,zlib]:x64-windows -> 12.2#20
Detecting compiler hash for triplet x64-windows...
Restored 0 packages from C:\Users\harry.ha\AppData\Local\vcpkg\archives in 312 us. Use --debug to see more details.
Starting package 1/1: libpq:x64-windows
Building package libpq[core,openssl,zlib]:x64-windows...
-- Using cached postgresql-12.2.tar.bz2.
-- Cleaning sources at D:/dev/vcpkg/buildtrees/libpq/src/tgresql-12-ef0b4889ed.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source D:/dev/vcpkg/downloads/postgresql-12.2.tar.bz2
-- Applying patch patches/windows/install.patch
-- Applying patch patches/windows/win_bison_flex.patch
-- Applying patch patches/windows/openssl_exe_path.patch
-- Applying patch patches/windows/Solution.patch
-- Applying patch patches/windows/MSBuildProject_fix_gendef_perl.patch
-- Applying patch patches/windows/msgfmt.patch
-- Applying patch patches/windows/python_lib.patch
-- Applying patch patches/windows/fix-compile-flag-Zi.patch
-- Applying patch patches/windows/minimize_install.patch
-- Using source at D:/dev/vcpkg/buildtrees/libpq/src/tgresql-12-ef0b4889ed.clean
-- Copying libpq source files to D:/dev/vcpkg/buildtrees/libpq/x64-windows-rel...
-- Copying libpq source files... done
CMake Deprecation Warning at scripts/ports.cmake:75 (message):
  vcpkg_apply_patches has been deprecated in favor of the `PATCHES` argument
  to `vcpkg_from_*`.
Call Stack (most recent call first):
  scripts/cmake/vcpkg_apply_patches.cmake:18 (z_vcpkg_deprecation_message)
  ports/libpq/portfile.cmake:127 (vcpkg_apply_patches)
  scripts/ports.cmake:142 (include)


-- Applying patch patches/windows/Solution_RELEASE.patch
-- Applying patch patches/windows/python3_build_RELEASE.patch
-- Patches applied!
-- Building libpq x64-windows-RELEASE...
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:127 (message):
    Command failed: D:/dev/vcpkg/downloads/tools/perl/5.32.1.1/perl/bin/perl.exe build.pl RELEASE libpq
    Working Directory: D:/dev/vcpkg/buildtrees/libpq/x64-windows-rel/src/tools/msvc
    Error code: 1
    See logs for more information:
      D:\dev\vcpkg\buildtrees\libpq\build-libpq-x64-windows-RELEASE-out.log

Call Stack (most recent call first):
  ports/libpq/portfile.cmake:223 (vcpkg_execute_required_process)
  scripts/ports.cmake:142 (include)


Error: Building package libpq:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `git pull` and `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  package: libpq[core,openssl,zlib]:x64-windows -> 12.2#20
  vcpkg version: 2021-11-02-af04ebf6274fd6f7a941bff4662b3955c64f6f42
  vcpkg-tool version: c973b499f 2021-11-14 (6 hours ago)

Additionally, attach any relevant sections from the log files above.

**Additional context**
I tried to use admin rights, install cmake, still failed. 

See attachment for log

[build-libpq-x64-windows-RELEASE-out.log](https://github.com/microsoft/vcpkg/files/7536050/build-libpq-x64-windows-RELEASE-out.log)

I'm getting nearly the same error message (but with visual studio 2022)
[build-libpq-x64-windows-DEBUG-err.log](https://github.com/microsoft/vcpkg/files/7537783/build-libpq-x64-windows-DEBUG-err.log)
[build-libpq-x64-windows-DEBUG-out.log](https://github.com/microsoft/vcpkg/files/7537784/build-libpq-x64-windows-DEBUG-out.log)
[build-libpq-x64-windows-RELEASE-err.log](https://github.com/microsoft/vcpkg/files/7537785/build-libpq-x64-windows-RELEASE-err.log)
[build-libpq-x64-windows-RELEASE-out.log](https://github.com/microsoft/vcpkg/files/7537786/build-libpq-x64-windows-RELEASE-out.log)


```
Building package libpq[core,openssl,zlib]:x64-windows...
-- Using cached postgresql-12.2.tar.bz2.
-- Cleaning sources at D:/dev/vcpkg/buildtrees/libpq/src/tgresql-12-ef0b4889ed.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source D:/dev/vcpkg/downloads/postgresql-12.2.tar.bz2
-- Applying patch patches/windows/install.patch
-- Applying patch patches/windows/win_bison_flex.patch
-- Applying patch patches/windows/openssl_exe_path.patch
-- Applying patch patches/windows/Solution.patch
-- Applying patch patches/windows/MSBuildProject_fix_gendef_perl.patch
-- Applying patch patches/windows/msgfmt.patch
-- Applying patch patches/windows/python_lib.patch
-- Applying patch patches/windows/fix-compile-flag-Zi.patch
-- Applying patch patches/windows/minimize_install.patch
-- Using source at D:/dev/vcpkg/buildtrees/libpq/src/tgresql-12-ef0b4889ed.clean
-- Copying libpq source files to D:/dev/vcpkg/buildtrees/libpq/x64-windows-rel...
-- Copying libpq source files... done
CMake Deprecation Warning at scripts/ports.cmake:75 (message):
  vcpkg_apply_patches has been deprecated in favor of the `PATCHES` argument
  to `vcpkg_from_*`.
Call Stack (most recent call first):
  scripts/cmake/vcpkg_apply_patches.cmake:18 (z_vcpkg_deprecation_message)
  ports/libpq/portfile.cmake:127 (vcpkg_apply_patches)
  scripts/ports.cmake:142 (include)


-- Applying patch patches/windows/Solution_RELEASE.patch
-- Applying patch patches/windows/python3_build_RELEASE.patch
-- Patches applied!
-- Building libpq x64-windows-RELEASE...
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:127 (message):
    Command failed: D:/dev/vcpkg/downloads/tools/perl/5.32.1.1/perl/bin/perl.exe build.pl RELEASE libpq
    Working Directory: D:/dev/vcpkg/buildtrees/libpq/x64-windows-rel/src/tools/msvc
    Error code: 1
    See logs for more information:
      D:\dev\vcpkg\buildtrees\libpq\build-libpq-x64-windows-RELEASE-out.log
      D:\dev\vcpkg\buildtrees\libpq\build-libpq-x64-windows-RELEASE-err.log

Call Stack (most recent call first):
  ports/libpq/portfile.cmake:223 (vcpkg_execute_required_process)
  scripts/ports.cmake:142 (include)


Error: Building package libpq:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `git pull` and `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  package: libpq[core,openssl,zlib]:x64-windows -> 12.2#20
  vcpkg version: 2021-11-02-af04ebf6274fd6f7a941bff4662b3955c64f6f42
  vcpkg-tool version: 334bafd7b 2021-11-15 (9 hours ago)

Additionally, attach any relevant sections from the log files above.
```

--- a/scripts/buildsystems/msbuild/applocal.ps1
+++ b/scripts/buildsystems/msbuild/applocal.ps1
@@ -58,7 +58,7 @@ function deployBinary([string]$targetBinaryDir, [string]$SourceDir, [string]$tar
             Write-Verbose "  ${targetBinaryName}: Copying $sourceBinaryFilePath"
             Copy-Item $sourceBinaryFilePath $targetBinaryDir
         }
-        if ($copiedFilesLog) { Add-Content $copiedFilesLog targetBinaryFilePath -Encoding UTF8 }
+        if ($copiedFilesLog) { Add-Content $copiedFilesLog $targetBinaryFilePath -Encoding UTF8 }
         if ($tlogFile) { Add-Content $tlogFile $targetBinaryFilePath -Encoding Unicode }
     } finally {
         if ($mtx) {

